File: solutions/raindrops/Ruby/raindrops.rb
module Raindrops
  FACTORS = { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }
  def self.convert(n)
    result = FACTORS.select { |k, _| n % k == 0 }.values.join
    result.empty? ? n.to_s : result
  end
end

File: solutions/raindrops/C++/raindrops.cpp
#include "raindrops.h"

namespace raindrops {
    std::string convert(unsigned int n) {
        std::string out = "";

        if (n % 3 == 0) {
            out += "Pling";
        }
        if (n % 5 == 0) {
            out += "Plang";
        }
        if (n % 7 == 0) {
            out += "Plong";
        }

        if (out != "") {
            return out;
        }
        return std::to_string(n);
    }
}
